### PR TITLE
feat: TPU *Create / *get / *create_with_script / *delete Samples

### PR DESCRIPTION
**Pull request by Thoughtseize1**
## Description

Created new Samples for TPU. Create, Get, Delete operations.
Also added Sample for creating TPU with Startup Script
Doc page -> [here](https://cloud.google.com/tpu/docs/managing-tpus-tpu-vm#gcloud)

| Region tag    | Sample                    |
|---------------|---------------------------|
| `tpu_vm_create` | `/tpu/create_tpu.py` |
| `tpu_vm_get`    | `/tpu/get_tpu.py`    |
| `tpu_vm_delete` | `/tpu/delete_tpu.py` |
| `tpu_vm_create_startup_script` | `/tpu/create_tpu_with_script.py` |

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [x] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Review by m-strzelczyk — CHANGES_REQUESTED** [review]
Looks good!

I forgot to tell you all, TPUs are a separate product from Compute (politics....). So please move the whole tpu folder, a level higher to the repository root.

You'll need to update `CODEOWNERS` file and `.github/blunderbuss.yml` file. Just copy the setting of `compute` for `tpu`.

**Comment by Thoughtseize1** [comment]
> Looks good!
> 
> I forgot to tell you all, TPUs are a separate product from Compute (politics....). So please move the whole tpu folder, a level higher to the repository root.
> 
> You'll need to update `CODEOWNERS` file and `.github/blunderbuss.yml` file. Just copy the setting of `compute` for `tpu`.

I've moved the TPU folder to the root of the repository. I updated the `CODEOWNERS` file and `.github/blunderbuss.yml`. Could you please review and confirm if everything is correct? @m-strzelczyk

File: tpu/test_tpu.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    except Exception as e:
+        print(f"Error during cleanup: {e}")
+
+
+def test_creating_tpu(tpu_instance: Node) -> None:
+    assert tpu_instance.state == Node.State.READY
+
+
+def test_creating_with_startup_script() -> None:
+    tpu_name_with_script = "script-ptu" + uuid.uuid4().hex[:5]
```
Comment (m-strzelczyk) [inline tpu/test_tpu.py:46]:
```suggestion
    tpu_name_with_script = "script-tpu-" + uuid.uuid4().hex[:5]
```

**Review by m-strzelczyk — APPROVED** [review]

File: tpu/create_tpu_with_script.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    sudo pip3 install --upgrade numpy >> /var/log/hello.log 2>&1
+    """
+    }
+    node = tpu_v2.Node()
+    # Adding metadata with startup script to the TPU node.
+    node.metadata = metadata
+    # Enabling external IPs for internet access from the TPU node.
+    node.network_config = tpu_v2.NetworkConfig(enable_external_ips=True)
+
+    # [END tpu_vm_create_startup_script]
```
Comment (m-strzelczyk) [inline tpu/create_tpu_with_script.py:49]: Maybe make this region tag span the whole node creation code. The gcloud example in docs is a full gcloud command, so we should have a full creation script as well IMO.

File: tpu/test_tpu.py (hunk 1 above)
Comment (Thoughtseize1) [reply tpu/test_tpu.py]: Done

File: tpu/create_tpu_with_script.py (hunk 2 above)
Comment (Thoughtseize1) [reply tpu/create_tpu_with_script.py]: I changed it to provide users with a full example of creating TPU with the startup script 👍

**Pull request merged**